src/python/pants/rules/core/run_test.py

```diff
@@ -15,7 +15,7 @@
 from pants.testutil.test_base import TestBase
 
 
-# TODO: Create a utility to mock GoalSubsystems.
+# TODO(#9141): replace this with a proper util to create `GoalSubsystem`s
```

Comment:
:D I appreciate you doing this and the other fixes (removing types from docstrings, using `PurePath`...). Thanks!